src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzeResultWriter.cs

```diff
@@ -49,19 +49,38 @@ await foreach (var ar in analyzeResultDefinitions)
                         {
                             Name = ar.Name,
                             SemanticVersion = ar.Version,
-                            InformationUri = ar.InformationURI,
-                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(a => new ReportingDescriptor()
-                            {
-                                Id = a.Key,
-                                FullDescription = new() { Text = a.First().RuleName, },
-                            }).ToList(),
+                            InformationUri = ar.InformationUri,
+                            Rules = analyzeResults.GroupBy(x => x.RuleId).Select(ExtractRule).ToList(),
                         },
                     },
                     Results = ExtractResults(analyzeResults),
                 };
             }
         }
 
+        private static ReportingDescriptor ExtractRule(IGrouping<string, AnalyzeResult> analyzeResults)
+        {
+            var analyzeResult = analyzeResults.First();
+
+            var rule = new ReportingDescriptor
+            {
+                Id = analyzeResult.RuleId,
+                HelpUri = analyzeResult.HelpUri
+            };
+
+            if (string.IsNullOrWhiteSpace(analyzeResult.FullDescription))
+            {
+                rule.FullDescription = new() { Text = analyzeResult.RuleName };
+            }
+            else
+            {
+                rule.Name = analyzeResult.RuleName;
+                rule.FullDescription = new() { Text = analyzeResult.FullDescription };
+            }
+
```

Comment:
instead you can just do `rule.FullDescription = new() { Text = !string.IsNullOrWhiteSpace(analyzeResult.FullDescription) ? analyzeResult.FullDescription : analyzeResult.RuleName };` no need of the `if..else`

Reply:
I removed the else, but we will still need the if.
Reason:
- FullDescription can be either the FullDescription or RuleName
- RuleName will only be RuleName if FullDescription is not null/empty (if fullDescription is null/empty, it means it is still using the default RuleName, which can contain multiple parts, spaces, etc.)

Reply:
makes sense. That way when we add FullDescription to all analyzers we can just remove the if here. Could you add a comment here so we will know this is temporary and will be removed in the subsequent PRs?